src/Illuminate/Console/Concerns/ConfiguresPrompts.php

```diff
@@ -167,4 +168,48 @@ protected function restorePrompts()
     {
         Prompt::setOutput($this->output);
     }
+
+    /**
+     * Validate the given prompt.
+     *
+     * @param  mixed  $value
+     * @param  mixed  $rules
+     * @return ?string
+     */
+    protected function validatePrompt($value, $rules)
+    {
+        if (! $rules) {
+            return null;
+        }
+
+        $field = 'answer';
+
+        if (is_array($rules) && ! array_is_list($rules)) {
+            [$field, $rules] = [key($rules), current($rules)];
+        }
+
+        return Validator::make([$field => $value], [$field => $rules], $this->messages(), $this->attributes())
+            ->errors()
+            ->first();
+    }
+
+    /**
+     * Get the validation messages.
+     *
+     * @return array
+     */
+    protected function messages()
+    {
+        return [];
+    }
+
+    /**
+     * Get the validation attributes.
+     *
+     * @return array
+     */
+    protected function attributes()
+    {
+        return [];
+    }
```

Comment:
I'm worried these method names are too generic in the context of a command.

I'm not sure how commonly people will want to customize these, but perhaps we could instead have a single method that receives `$data` and `$rules` and returns a `Validator` instance. The method could then be overridden for any required customizations.

Reply:
The original intention was to be consistent with `FormRequest::messages()` and `FormRequest::attributes()` but I do agree with your concerns 👍 

I extracted the validator instantiation logic to the `getValidatorInstance()` method (consistent with `FormRequest`) and renamed the messages and attributes methods to `validationMessages()` and `validationAttributes()` so that they are not too generic.

I think this may be a good compromise to have the flexibility to totally customise the Validator and also the ability to simply customise the errors/attributes without the need of manually instantiating the Validator. However if you prefer to just have the `getValidatorInstance()` method, I'll remove the other two 👍

Reply:
Nice! I've often thought it would be cool to add validation rules to command arguments and options as well, and I _think_ these methods could be used for that too.